src/components/HOC/TodoMenuAuthCheck.tsx

```diff
@@ -0,0 +1,18 @@
+import useUserInfo from "../../hooks/useUserInfo";
+
+const withConditionalAuth = (Auth, NonAuth) => {
+  const route = () => {
```

Comment:
It seems like we don't need the route function and can have hooks inside the withConditionalAuth like you have in `withAuthRedirect`